ray-operator/controllers/ray/utils/httpproxy_httpclient.go

```diff
@@ -50,7 +50,11 @@ func (r *RayHttpProxyClient) SetHostIp(hostIp, podNamespace, podName string, por
 
 // CheckProxyActorHealth checks the health status of the Ray Serve proxy actor.
 func (r *RayHttpProxyClient) CheckProxyActorHealth(ctx context.Context) error {
-	resp, err := r.client.Get(r.httpProxyURL + RayServeProxyHealthPath)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, r.httpProxyURL+RayServeProxyHealthPath, nil)
```

Comment:
Note: "You should use noctx if sending http request in your library. Passing context.Context enables library user to cancel http request, getting trace information and so on."

https://github.com/sonatard/noctx